src/server/pkg/log/log.go

```diff
@@ -228,9 +228,19 @@ func topLevelService(fullyQualifiedService string) string {
 	return tokens[0]
 }
 
-type prettyFormatter struct{}
+// PrettyFormatter formats logrus log messages like so:
+// ```
+// 2019-02-11T16:02:02Z INFO pfs.API.InspectRepo {"request":{"repo":{"name":"images"}}} []
+// ```
+type PrettyFormatter struct{}
```

Comment:
Empty structs like this are a totally reasonable way to implement an interface. There is a [slightly nicer way](https://golang.org/src/net/http/server.go?s=73427:73498#L1960) (imo) that the standard library uses in the http server code. More just an fyi than anything, I'm happy to approve either method.

Reply:
Sorry, I'm not following. I can't use a closure here because I need to implement an interface for logrus.

Reply:
Extended your example to show what I'm talking about: https://play.golang.org/p/6IBl1z7WooI

The main difference is that rather than using a `struct{}` as the receiver for the func that satisfies the interface you can actually use a func itself as the receiver. Because funcs are first class objects in go they can act as receivers. Like I said, not a required change at all.

Reply:
<img width="290" alt="screen shot 2019-02-13 at 4 45 59 pm" src="https://user-images.githubusercontent.com/127386/52746173-e2f47f80-2fae-11e9-9bfb-cdfbed978828.png">

Reply:
Changed it to a closure